js/lib/findForm.js

```diff
@@ -1,255 +1,8 @@
-                if (pw1 === pw2) {
-                    // Treat as if 1 pw field
-                    newPasswordField = pwFields[0].element;
-                    oldPasswordField = null;
-                } else {
-                    // Just assume that the 2nd password is the new password
-                    oldPasswordField = pwFields[0].element;
-                    newPasswordField = pwFields[1].element;
-                }
-            }
-
-            return [usernameField, newPasswordField, oldPasswordField];
-        },
-        log: function (str) {
-            if(settings.debug){
-                console.log(str);
-            }
-        }
-    };
-}();
-
-function getLoginFields(isSubmission) {
-    var forms = document.forms;
-    var loginForms = [];
-
-    for (var i = 0; i < forms.length; i++) {
-        var form = forms[i];
-        var result = formManager.getFormFields(form, isSubmission);
-        var usernameField = result[0];
-        var passwordField = result[1];
-        // Need a valid password field to do anything.
-        if (passwordField === null){
-            continue;
-        }
-
-        var res = [usernameField, passwordField];
-        if(result[2]){
-            res.push(result[2]);
-        } else {
-            res.push(null);
-        }
-        loginForms.push(res);
-    }
-    return loginForms;
-}
-
-function getFormFromElement(elem) {
-    if(elem) {
-        while (elem.parentNode) {
-            if (elem.parentNode.nodeName.toLowerCase() === "form") {
-                return elem.parentNode;
-            }
-            elem = elem.parentNode;
-        }
-    }
-}
-
-function dispatchEvents(element){
-    var eventNames = [ 'click', 'focus', 'keypress', 'keydown', 'keyup', 'input', 'blur', 'change' ];
-    eventNames.forEach(function(eventName) {
-        element.dispatchEvent(new Event(eventName, {"bubbles":true}));
-    });
-}
-
 function fillPassword(user, password) {
-    var loginFields = getLoginFields();
-    for (var i = 0; i < loginFields.length; i++) {
-        if(user && loginFields[i][0]){
-            loginFields[i][0].value = user;
-            if(loginFields[i][0].offsetParent) {
-                dispatchEvents(loginFields[i][0]);
-            }
-        }
-        if(password && loginFields[i][1]) {
-            loginFields[i][1].value = password;
-            if(loginFields[i][1].offsetParent) {
-                dispatchEvents(loginFields[i][1]);
-            }
-        }
-        if(password && loginFields[i][2]) {
-            loginFields[i][2].value = password;
-            if(loginFields[i][2].offsetParent) {
-                dispatchEvents(loginFields[i][2]);
-            }
-        }
+    if(user){
+        document.querySelectorAll('input[type=text], input[type=email]').forEach(function(x){x.value=user;});
+    }
+    if(password) {
+        document.querySelectorAll('input[type=password]').forEach(function(x){ x.value=password;});
```

Comment:
I'm missing the events (see `dispatchEvents`) for example. Many sites rely on those functions being called.
Also, do you have tested it throughly?